src/riak_ensemble_client.erl

```
@@ -105,3 +125,12 @@ translate(Result) ->
             %%       Perhaps build this logic into peer to return {error, notfound}?
             Result
     end.
+
+-spec maybe(node(), fun()) -> client_reply().
+maybe(Node, Fun) ->
+    case catch rpc:call(Node, riak_ensemble_manager, enabled, []) of
```

I agree. This is a fail fast scenario. Fail fast if local node isn't enabled, otherwise just send the request like we already do prior to this PR and let the request timeout.